File: tests/adapters/javascript/goldens/imports/strip_external.js
/**
 * JavaScript module for testing import optimization.
 */

// External library imports (should be considered external)
// … 16 imports omitted

// Scoped package imports (external)
// … 4 imports omitted

// Node.js built-in modules (external/standard library)
// … 7 imports omitted

// Local/relative imports (should be considered local)
import { UserService } from './services/user-service.js';
import { DatabaseConnection } from './database/connection.js';
import { ValidationError, NetworkError } from './errors.js';
import { formatDate, parseJson } from './utils/helpers.js';
import { ApiResponse, UserModel, PostModel } from './types.js';

// Relative imports with different depth levels
import { SharedUtility } from '../shared/utilities.js';
import { CoreModule } from '../../core/core.module.js';
import { AppConfig } from '../../../config/app.config.js';

// Import with aliasing
import { Logger as AppLogger } from './utils/logger.js';
import { Config as AppConfig2 } from './config/app.config.js';
import { default as HttpClient } from './http/client.js';

// Mixed import styles on single line
// … 7 imports omitted

// Long import lists (candidates for summarization)
import {
    createUser,
    updateUser,
    deleteUser,
    getUserById,
    getUserByEmail,
    getUsersByRole,
    getUsersWithPagination,
    activateUser,
    deactivateUser,
    resetUserPassword,
    changeUserRole,
    validateUserPermissions
} from './services/user-operations.js';

import {
    validateEmail,
    validatePassword,
    validatePhoneNumber,
    validatePostalCode,
    validateCreditCard,
    sanitizeInput,
    formatCurrency,
    formatPhoneNumber,
    generateSlug,
    createHash,
    verifyHash
} from '../utils/validation-helpers.js';

// Dynamic imports (should be preserved as-is)
const dynamicModule = async () => {
    const { default: chalk } = await import('chalk');
    return chalk;
};

// Conditional imports
let csvParser;
try {
    csvParser = require('csv-parser');
} catch (error) {
    console.warn('csv-parser not available');
}

export class ImportTestService {
    constructor(userService, dbConnection, logger) {
        this.userService = userService;
        this.dbConnection = dbConnection;
        this.logger = logger;
    }

    async processData(data) {
        // Using external libraries
        const processed = lodash.map(data, item => ({
            id: uuidv4(),
            timestamp: moment().toISOString(),
            ...item
        }));

        // Using local utilities
        const validated = processed.map(item =>
            validateEmail(item.email) ? item : null
        ).filter(Boolean);

        // Using Node.js built-ins
        const filePath = path.join(__dirname, 'output.json');
        await promisify(fs.writeFile)(filePath, JSON.stringify(validated));

        return {
            success: true,
            data: validated,
            timestamp: formatDate(new Date())
        };
    }

    async makeHttpRequest(url) {
        try {
            // Using axios
            const response = await axios.get(url, {
                timeout: 5000,
                headers: {
                    'User-Agent': 'ImportTestService/1.0'
                }
            });

            return response.data;
        } catch (error) {
            this.logger.error('HTTP request failed', error);
            throw new NetworkError('Request failed');
        }
    }
}

// Re-exports (should be handled appropriately)
export { UserService } from './services/user-service.js';
export { DatabaseConnection } from './database/connection.js';
export * from './types.js';

// Default export
export default ImportTestService;

File: tests/adapters/kotlin/goldens/imports/strip_external.kt
/**
 * Kotlin module for testing import optimization.
 */

package com.example.imports

// … 22 imports omitted (24 lines)

// Local/relative imports (should be considered local)
import com.example.imports.services.UserService
import com.example.imports.database.DatabaseConnection
import com.example.imports.errors.ValidationError
import com.example.imports.errors.NetworkError
import com.example.imports.utils.helpers.formatDate
import com.example.imports.utils.helpers.parseJson
import com.example.imports.types.ApiResponse
import com.example.imports.types.UserModel
import com.example.imports.types.PostModel

// Relative imports with different depth levels
import com.example.shared.SharedUtility
import com.example.core.CoreModule
import com.example.config.AppConfig

// Import with aliasing
import com.example.imports.utils.logger.Logger as AppLogger
import com.example.imports.config.Config as AppConfig2
import com.example.imports.http.HttpClient as CustomHttpClient

// Star imports
import com.example.imports.extensions.*
import com.example.imports.constants.*

// … 39 imports omitted (40 lines)

// Local imports with long lists
import com.example.imports.services.createUser
import com.example.imports.services.updateUser
import com.example.imports.services.deleteUser
import com.example.imports.services.getUserById
import com.example.imports.services.getUserByEmail
import com.example.imports.services.getUsersByRole
import com.example.imports.services.getUsersWithPagination
import com.example.imports.services.activateUser
import com.example.imports.services.deactivateUser
import com.example.imports.services.resetUserPassword
import com.example.imports.services.changeUserRole
import com.example.imports.services.validateUserPermissions

import com.example.utils.validation.validateEmail
import com.example.utils.validation.validatePassword
import com.example.utils.validation.validatePhoneNumber
import com.example.utils.validation.validatePostalCode
import com.example.utils.validation.validateCreditCard
import com.example.utils.validation.sanitizeInput
import com.example.utils.validation.formatCurrency
import com.example.utils.validation.formatPhoneNumber
import com.example.utils.validation.generateSlug
import com.example.utils.validation.createHash
import com.example.utils.validation.verifyHash

class ImportTestService(
    private val userService: UserService,
    private val dbConnection: DatabaseConnection,
    private val logger: AppLogger
) : KoinComponent {
    
    suspend fun processData(data: List<Any>): ApiResponse<List<Any>> {
        // Using external libraries
        val processed = data.map { item ->
            mapOf(
                "id" to UUID.randomUUID(),
                "timestamp" to Instant.now().toString(),
                "item" to item
            )
        }
        
        // Using local utilities
        val validated = processed.mapNotNull { item ->
            val email = item["email"] as? String
            if (email != null && validateEmail(email)) item else null
        }
        
        // Using Kotlin standard library
        val filePath = Path("/output.json")
        
        return ApiResponse(
            success = true,
            data = validated,
            timestamp = formatDate(LocalDateTime.now())
        )
    }
    
    suspend fun makeHttpRequest(url: String): String? {
        return try {
            // Using Ktor client
            val client = HttpClient()
            val response: HttpResponse = client.get(url) {
                timeout {
                    requestTimeoutMillis = 5000
                }
                headers {
                    append("User-Agent", "ImportTestService/1.0")
                }
            }
            
            response.bodyAsText()
        } catch (e: Exception) {
            logger.error("HTTP request failed", e)
            throw NetworkError("Request failed")
        }
    }
    
    fun serializeData(data: Any): String {
        // Using Gson
        val gson = GsonBuilder()
            .setPrettyPrinting()
            .create()
        
        return gson.toJson(data)
    }
}

// Annotated classes using validation imports
@Serializable
data class ValidatedUser(
    @NotNull
    @NotEmpty
    val id: String,
    
    @NotBlank
    @Size(min = 2, max = 100)
    val name: String,
    
    @Email
    @NotEmpty
    val email: String,
    
    @Positive
    val age: Int?,
    
    @Pattern(regexp = "^\\+?[1-9]\\d{1,14}$")
    val phone: String?
)

// Function using multiple imports
suspend fun processWithCoroutines(items: List<String>): Flow<String> = flow {
    items.forEach { item ->
        delay(100)
        emit(item.uppercase())
    }
}.flowOn(Dispatchers.Default)

// Koin module definition
val appModule = module {
    single { DatabaseConnection(get()) }
    singleOf(::UserService) bind AppLogger::class
    factory { ImportTestService(get(), get(), get()) }
}

// Default export
fun main() {
    startKoin {
        modules(appModule)
    }
    
    val service: ImportTestService by inject()
    println("Service initialized")
}
